scripts/cli/handler/parse.py

```diff
@@ -0,0 +1,377 @@
+    f.write(after)
+
+def __create_database_and_clean(args):
+  #--- Helper variables ---#
+
+  projectdir = join(args.workdir, args.name)
+  dblogfile = join(projectdir, 'dblog')
+
+  #--- Database operations ---#
+
+  print __current_time(), 'Creating new database...'
+
+  __collect_sql(projectdir)
+
+  util.pgsql_cmd(projectdir, args, 'dropdb', args.name)
+
+  res = util.pgsql_cmd(
+    projectdir, args,
+    'createdb', '-E', 'SQL_ASCII', '-T', 'template0', '-l', 'C', args.name)
+
+  if res != 0:
+    sys.stderr.write(
+      'createdb returned with ' + str(res) + '!\nSee ' + dblogfile +
+      ' for details!\n')
+    sys.exit(1)
+
+  res = util.pgsql_cmd(
+    projectdir, args, 'psql', '-f', 'parse.sql', '-d', args.name)
+
+  if res != 0:
+    sys.stderr.write(
+      'Failed to create the database schema (psql returned with ' + str(res) +
+      ')!\nSee ' + dblogfile + ' for details!\n')
+    sys.exit(1)
+
+  #--- Clean ---#
+
+  try:
+    shutil.rmtree(join(projectdir, 'data'))
+    os.remove(join(projectdir, 'codechecker_output'))
+    os.remove(join(projectdir, 'parselog'))
+    os.remove(join(projectdir, 'buildlog'))
+    for filename in os.listdir(projectdir):
+      if filename.startswith('core') or filename.startswith('crash_'):
+        os.remove(join(projectdir, filename))
+  except:
+    pass
+
+  config_file_handler = ConfigFileHandler(join(args.workdir, 'workspace.cfg'))
+  config_file_handler.remove(args.name)
+  config_file_handler.save()
+
+  print __current_time(), 'Done'
+
+def __run_after_sql_script(args):
+  print __current_time(), 'Creating indexes and constraints...'
+
+  projectdir = join(args.workdir, args.name)
+
+  res = util.pgsql_cmd(
+    projectdir, args, 'psql', '-f', 'after.sql', '-d', args.name)
+
+  if res == 0:
+    res = util.pgsql_cmd(
+      projectdir, args, 'psql', '-c', 'VACUUM ANALYZE;', '-d', args.name)
+
+  if res != 0:
+    sys.stderr.write(
+      'Failed to create the database indexes (psql returned with ' + str(res) +
+      ')!\nSee ' + dblogfile + ' for details!\n')
+    sys.exit(1)
+
+  print __current_time(), 'Done'
+
+def parse_handler(args):
+  signal.signal(signal.SIGINT, interrupt_parse_handler)
+
+  #--- Print database connection data ---#
+
+  print '--- Database connection ---'
+  print 'Database host: ' +     args.dbhost
+  print 'Database port: ' + str(args.dbport)
+  print 'Database user: ' +     args.dbuser
+  print 'Database pass: ' +     args.dbpass
+  print '---------------------------'
+
+  #--- Create workdir and project directory ---#
+
+  projectdir = join(args.workdir, args.name)
+
+  if not os.path.exists(projectdir):
+    try:
+      os.makedirs(projectdir)
+    except OSError:
+      sys.stderr.write('Permission denied to create ' + projectdir + '\n')
+      sys.exit(1)
+  elif not args.force:
+    sys.stderr.write(projectdir + ' already exists. Use -f to force reparsing!\n')
+    sys.exit(1)
+  else:
+    if args.logfile:
```

Comment:
If we set the `--logfile` parameter, it will use the build.json from project directory so we don't allow to remove it from the project directory.